src/attr/_make.py

```diff
@@ -680,6 +680,10 @@ def fmt_setter_with_converter(attr_name, value_var):
                                                     a.name))
         names_for_globals[val_name] = a.validator
         names_for_globals[attr_name] = a
+    lines.extend([
+        "func = getattr(self, '__post_init__', None)",
```

Comment:
This is approach means that `attrs` gets universally slower for everyone.  `attrs` is explicitly static in its approach.  We need to check for the existence of the method when assembling the class and call it unconditionally if it’s present.